tests/test_cases/test_array/test_array.py

```diff
@@ -71,54 +33,54 @@ async def test_read_write(dut):
     await Timer(10, "ns")
 
     tlog.info("Checking Generics/Parameters:")
-    _check_logic(tlog, dut.param_logic, 1)
-    _check_logic(tlog, dut.param_logic_vec, 0xDA)
+    assert dut.param_logic.value == 1
+    assert dut.param_logic_vec.value == 0xDA
 
     if LANGUAGE in ["vhdl"]:
-        _check_int(tlog, dut.param_bool, 1)
-        _check_int(tlog, dut.param_int, 6)
-        _check_real(tlog, dut.param_real, 3.14)
-        _check_int(tlog, dut.param_char, ord("p"))
-        _check_str(tlog, dut.param_str, b"ARRAYMOD")
+        assert dut.param_bool.value == 1
+        assert dut.param_int.value == 6
+        assert dut.param_real.value == 3.14
+        assert dut.param_char.value == ord("p")
+        assert dut.param_str.value == b"ARRAYMOD"
 
         if not cocotb.SIM_NAME.lower().startswith("riviera"):
-            _check_logic(tlog, dut.param_rec.a, 0)
-            _check_logic(tlog, dut.param_rec.b[0], 0)
-            _check_logic(tlog, dut.param_rec.b[1], 0)
-            _check_logic(tlog, dut.param_rec.b[2], 0)
-            _check_logic(tlog, dut.param_cmplx[0].a, 0)
-            _check_logic(tlog, dut.param_cmplx[0].b[0], 0)
-            _check_logic(tlog, dut.param_cmplx[0].b[1], 0)
-            _check_logic(tlog, dut.param_cmplx[0].b[2], 0)
-            _check_logic(tlog, dut.param_cmplx[1].a, 0)
-            _check_logic(tlog, dut.param_cmplx[1].b[0], 0)
-            _check_logic(tlog, dut.param_cmplx[1].b[1], 0)
-            _check_logic(tlog, dut.param_cmplx[1].b[2], 0)
+            assert dut.param_rec.a.value == 0
+            assert dut.param_rec.b[0].value == 0
+            assert dut.param_rec.b[1].value == 0
+            assert dut.param_rec.b[2].value == 0
+            assert dut.param_cmplx[0].a.value == 0
+            assert dut.param_cmplx[0].b[0].value == 0
+            assert dut.param_cmplx[0].b[1].value == 0
+            assert dut.param_cmplx[0].b[2].value == 0
+            assert dut.param_cmplx[1].a.value == 0
+            assert dut.param_cmplx[1].b[0].value == 0
+            assert dut.param_cmplx[1].b[1].value == 0
+            assert dut.param_cmplx[1].b[2].value == 0
 
     tlog.info("Checking Constants:")
```

Comment:
Since the asserts do not print anything for the passing case, these info strings could be deleted as well.